### ISSUE TITLE
This extension does not work with the latest version of PHP.

### ISSUE THREAD

**Issue by maestroprog** [krakjoe/uopz#122]
After starting any line of code with the extension turned on, the PHP hangs up.  Reproduced on PHP 7.3.16 and PHP 7.4.3
The last release 19 Mar 2020 https://www.php.net/archive/2020.php

**Comment by maestroprog** [comment]
```
# php-config
Usage: /usr/bin/php-config [OPTION]
Options:
  --prefix            [/usr]
  --includes          [-I/usr/include/php/20180731 -I/usr/include/php/20180731/main -I/usr/include/php/20180731/TSRM -I/usr/include/php/20180731/Zend -I/usr/include/php/20180731/ext -I/usr/include/php/20180731/ext/date/lib ]
  --ldflags           [-L/usr/lib/php/20180731 ]
  --libs              [-lcrypt   -largon2 -lresolv -lcrypt -lz -lrt -lm -ldl -lnsl  -lxml2 -lssl -lcrypto -lpcre2-8 -lsodium -lcrypt -largon2 -lcrypt ]
  --extension-dir     [/usr/lib/php/20180731]
  --include-dir       [/usr/include/php/20180731]
  --man-dir           [/usr/share/man]
  --php-binary        [/usr/bin/php7.3]
  --php-sapis         [cli ]
  --phpapi            [20180731]
  --configure-options [--includedir=/usr/include --mandir=/usr/share/man --infodir=/usr/share/info --disable-silent-rules --libdir=/usr/lib/x86_64-linux-gnu --libexecdir=/usr/lib/x86_64-linux-gnu --disable-maintainer-mode --disable-dependency-tracking --prefix=/usr --enable-cli --disable-cgi --disable-phpdbg --with-config-file-path=/etc/php/7.3/cli --with-config-file-scan-dir=/etc/php/7.3/cli/conf.d --build=x86_64-linux-gnu --host=x86_64-linux-gnu --config-cache --cache-file=/build/php7.3-TxKPzY/php7.3-7.3.16/config.cache --libdir=${prefix}/lib/php --libexecdir=${prefix}/lib/php --datadir=${prefix}/share/php/7.3 --program-suffix=7.3 --sysconfdir=/etc --localstatedir=/var --mandir=/usr/share/man --disable-all --disable-debug --disable-rpath --disable-static --with-pic --with-layout=GNU --without-pear --enable-filter --with-openssl=yes --with-password-argon2=/usr --with-pcre-regex=/usr --enable-hash --with-mhash=/usr --enable-libxml --enable-session --with-sodium --with-system-tzdata --with-zlib=/usr --with-zlib-dir=/usr --enable-dtrace --enable-pcntl --with-libedit=shared,/usr build_alias=x86_64-linux-gnu host_alias=x86_64-linux-gnu CFLAGS=-g -O2 -fdebug-prefix-map=/build/php7.3-TxKPzY/php7.3-7.3.16=. -fstack-protector-strong -Wformat -Werror=format-security -O2 -Wall -pedantic -fsigned-char -fno-strict-aliasing -g]
  --version           [7.3.16-1+ubuntu18.04.1+deb.sury.org+1]
  --vernum            [70316]
```


```
# php -v
PHP 7.3.16-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Mar 20 2020 13:51:46) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.16, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.16-1+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.9.3, Copyright (c) 2002-2020, by Derick Rethans
```


```
# php -i | grep uopz
/etc/php/7.3/cli/conf.d/30-uopz.ini
uopz
uopz support => enabled
uopz.disable => 0 => 0
uopz.exit => 0 => 0
```

```
# php -m
[PHP Modules]
bcmath
calendar
Core
ctype
curl
date
dom
exif
fileinfo
filter
ftp
gettext
hash
iconv
igbinary
imagick
json
libxml
mbstring
openssl
pcntl
pcre
PDO
pdo_pgsql
pgsql
Phar
posix
readline
redis
Reflection
session
shmop
SimpleXML
soap
sockets
sodium
SPL
standard
sysvmsg
sysvsem
sysvshm
tokenizer
uopz
wddx
xdebug
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Xdebug
Zend OPcache
```

**Comment by maestroprog** [comment]
> After starting any line of code with the extension turned on, the PHP hangs up.

Wrong, not on any line of code. But on any of function call. For example: `var_dump`,`print_r`,`var_export`,`array_intersect`...
Also, reproduced from command line:
`php -r 'array_intersect([1],[2]);'`

**Comment by cmb69** [comment]
Did you try without Xdebug?

**Comment by maestroprog** [comment]
Yes, I did it. Results: It is works, perhaps xdebug with uopz are in conflict.

**Comment by rybakit** [comment]
I confirm that `xdebug 2.9.4` and `uopz 6.1.1` do not work together, downgrading `uopz` to `6.1.0` solves the problem though.

**Comment by cmb69** [comment]
Confirmed. The culprit is the fix (d9785ce7357fae2a49ba5f00a6705cd75d6fbdcd) for #110 and #111. Since Xdebug 2.9.4 has fixed that long standing issue of not properly saving other OPcode handlers, that fix doesn't work anymore. Maybe it's best to revert that commit, and to document that Xdebug before 2.9.4 is not supported.

**Comment by JParkinson1991** [comment]
Just want to provide my findings, I came across this issue when trying to run with the following versions:

```
PHP: 7.4.6
Xdebug: 2.9.5
uopz: 6.1.1
```

Downgrading uopz to 6.1.0 fixed the issues.

**Comment by elyran** [comment]
Same issue.

```
PHP 7.3.13 (cli) (built: Dec 18 2019 23:16:14) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.13, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.13, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.9.6, Copyright (c) 2002-2020, by Derick Rethans
```

Downgrading to 6.1.0 solved the issue.

**Comment by krakjoe** [comment]
@cmb69 as you're familiar with these changes, can you do whatever is necessary in code and docs to fix and close this issue please ?

**Comment by cmb69** [comment]
This has apparently already been fixed with 1f30c82152d430c15a2a2b40fd463838bfabad69. I'm going to update the PHP manual.
